### DISCUSSION TITLE
Way to hide popover/tooltip when anchor is offscreen?

### DISCUSSION THREAD

**Discussion by HelveticaScenario** [ariakit/ariakit#2190] [Questions] [answered]
By default when a popover or tooltip is open, if the anchor is scrolled away to the left or right the dialog remains visible even after the anchor is gone. is there a way to disable this behavior? Here's a [codesandbox](https://codesandbox.io/s/falling-water-667vo4) to show what I mean.

**Comment by diegohaz** [comment, marked as answer]
There's no such feature, but you can achieve that behavior with `IntersectionObserver`: https://codesandbox.io/s/ecstatic-danny-hxty0h?file=/src/tooltip.tsx

**Reply by HelveticaScenario** [reply]
thank you, that seems to work. is there any plan/interest in adding this behavior to ariakit? I noticed that floating UI has a [shift limiter](https://floating-ui.com/docs/shift#limiter), could that do it?
